### PR TITLE
clippy: Fix assorted warnings in `components/`

### PR DESCRIPTION
**Pull request by eerii**
Continuation to #31500, fixes clippy warnings in the rest of `components/` modules (excluding all those in `script` and the ones about safety comments and result types).

It includes the following allows that seem to be reasonable:
- `upper_case_acronyms`: in the case of `WGPU` I think it looks better and more correct than `Wgpu`.
- `wrong_self_convention`: in `compositing/compositor.rs`, a function named `from_webrender` is used to turn `WebRenderPipelineId` into `PipelineId`. I can't think of a name that better represents this conversion, if anyone has ideas for this please do say them.
- `large_enum_variant`: same as in #31610

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #31500
- [x] These changes do not require tests because they do not modify functionality, they only fix lint errors.

### CODE REVIEW

File: components/webdriver_server/lib.rs [hunk 1]
```diff
@@ -189,51 +189,51 @@ struct Handler {
...
 #[derive(Clone, Copy, Debug, PartialEq)]
-enum ServoExtensionRoute {
-    GetPrefs,
-    SetPrefs,
-    ResetPrefs,
+enum ServoExtensionRoutePrefs {
+    Get,
+    Set,
+    Reset,
 }
```
Comment (mrobinson) [inline components/webdriver_server/lib.rs:196]: Hrm...I think these should probably stay. I'm not sure that `ServoExtensionRoutePrefs` makes sense...it's just that all of the ExtensionRoutes are currently pref related at the moment.

File: components/webdriver_server/lib.rs [hunk 2]
```diff
@@ -189,51 +189,51 @@ struct Handler {
...
 
 #[derive(Clone, Debug, PartialEq)]
-enum ServoExtensionCommand {
-    GetPrefs(GetPrefsParameters),
-    SetPrefs(SetPrefsParameters),
-    ResetPrefs(GetPrefsParameters),
+enum ServoExtensionCommandPrefs {
+    Get(GetPrefsParameters),
+    Set(SetPrefsParameters),
+    Reset(GetPrefsParameters),
```
Comment (mrobinson) [inline components/webdriver_server/lib.rs:228]: I would say the same here.

File: components/webgpu/lib.rs [hunk 3]
```diff
@@ -330,6 +331,12 @@ impl WebGPU {
     }
 }
 
+type WGPUBufferMaps<'a> =
+    HashMap<id::BufferId, Rc<BufferMapInfo<'a, Option<WebGPUResponseResult>>>>;
+type WGPUPresentBufferMaps<'a> =
+    HashMap<id::BufferId, Rc<BufferMapInfo<'a, (Option<ErrorScopeId>, WebGPURequest)>>>;
```
Comment (mrobinson) [inline components/webgpu/lib.rs:337]:
```suggestion
type WebGPUBufferMaps<'a> =
    HashMap<id::BufferId, Rc<BufferMapInfo<'a, Option<WebGPUResponseResult>>>>;
type WebGPUPresentBufferMaps<'a> =
    HashMap<id::BufferId, Rc<BufferMapInfo<'a, (Option<ErrorScopeId>, WebGPURequest)>>>;
```

Just for consistency with the rest of the code.

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: components/webdriver_server/lib.rs (hunk 1 above)
Comment (eerii) [reply components/webdriver_server/lib.rs]: You're right, it was cleaner originally. I reverted it back and added an allow.

**Review by mrobinson — APPROVED** [review]

**Pull request merged**